packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -260,8 +260,14 @@ let interruptedBy: Fiber | null = null;
 
 // Do not decrement interaction counts in the event of suspense timeouts.
 // This would lead to prematurely calling the interaction-complete hook.
+// Instead we wait until the suspended promise has resolved.
 let suspenseDidTimeout: boolean = false;
 
+// Do not remove the current interactions from the priority map on commit.
+// This covers a special case of sync rendering with suspense.
+// In this case we leave interactions in the Map until the suspended promise resolves.
+let preservePendingInteractionsOnCommit: boolean = true;
```

Comment:
This approach causes some interactions to get lumped together in the case of a suspended+sync root with a high priority interruption, but I think that might be acceptable?

I'd be happy to discuss alternatives!